google/cloud/storage/async/client.h

```diff
@@ -483,14 +483,23 @@ class AsyncClient {
    */
   template <typename... Args>
   future<StatusOr<std::pair<AsyncWriter, AsyncToken>>> StartBufferedUpload(
-      std::string const& bucket_name, std::string const& object_name,
-      Args&&... args) {
-    (void)bucket_name;
-    (void)object_name;
+      std::string bucket_name, std::string object_name, Args&&... args) {
     auto options = SpanOptions(std::forward<Args>(args)...);
-    (void)options;
-    return make_ready_future(StatusOr<std::pair<AsyncWriter, AsyncToken>>(
-        Status(StatusCode::kUnimplemented, "TODO(#13385)")));
+    return connection_
```

Comment:
Can we mock this in a unit test?

Reply:
Done.  I will add the unit tests for other functions too see #13419